flink-tests/src/test/java/org/apache/flink/test/javaApiOperators/CoGroupITCase.java

```diff
@@ -301,6 +309,51 @@ public void testCoGroupWithMultipleKeyFieldsWithFieldSelector() throws Exception
 	}
 
 	@Test
+	public void testCoGroupWithMultipleKeyFieldsWithFieldSelector2() throws Exception {
+		/*
+		 * UDF Join on tuples with multiple key field positions and same customized distribution
+		 */
+
+		final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+
+		DataSet<Tuple5<Integer, Long, Integer, String, Integer>> ds1 = CollectionDataSets.get5TupleDataSet(env)
+				.map(new MapFunction<Tuple5<Integer, Long, Integer, String, Long>, Tuple5<Integer, Long, Integer, String, Integer>>() {
```

Comment:
Can you keep the original data type, i.e., not convert the last field to Integer? 
1) It will check if the range partitioning handles different types.
2) it will be more concise (no map functions + no additional CoGroupFunction).